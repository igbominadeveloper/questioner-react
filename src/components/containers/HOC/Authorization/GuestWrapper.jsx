import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';

import { getItem } from '../../../../utils/helpers';
export const GuestWrapper = ({
  component: Component,
  isAuthenticated,
  ...rest
}) => {
  return (
    <Route
      {...rest}
      render={props =>
        !isAuthenticated ? <Component {...props} /> : <Redirect to="/" />
      }
    />
  );
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.token !== null,
});

export default connect(mapStateToProps)(withRouter(GuestWrapper));
