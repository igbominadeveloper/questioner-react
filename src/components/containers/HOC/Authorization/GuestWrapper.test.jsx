import React from 'react';
import { shallow } from 'enzyme';

import { GuestWrapper } from './GuestWrapper';

const component = shallow(<GuestWrapper />);

describe('GuestWrapper', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });
});
