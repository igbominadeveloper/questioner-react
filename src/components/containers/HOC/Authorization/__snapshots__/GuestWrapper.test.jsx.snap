// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestWrapper renders without errors 1`] = `
<Route
  render={[Function]}
/>
`;
